src/source/Common/EcsCredentialProvider.c

```diff
@@ -0,0 +1,290 @@
+/**
+ * Kinesis Video Producer ECS based Credential Provider
+ */
+#define LOG_CLASS "EcsCredentialProvider"
+#include "Include_i.h"
+
+STATUS getHostPort(PCHAR pUrl, PCHAR* ppStart, PCHAR* ppEnd)
```

Comment:
Generally, all of the private aux functionality follows the public API in a source file

Reply:
Done